backend/alembic/versions/0034_virtual_collections_db_view.py

```diff
@@ -0,0 +1,306 @@
+                    SELECT
+                        r.id as rom_id,
+                        r.path_cover_s as path_cover_s,
+                        r.path_cover_l as path_cover_l,
+                        jsonb_array_elements_text(igdb_metadata -> 'genres') as collection_name,
+                        'genre' as collection_type
+                    FROM roms r
+                    WHERE igdb_metadata->'genres' IS NOT NULL
+                ),
+                franchises_collection AS (
+                    SELECT
+                        r.id as rom_id,
+                        r.path_cover_s as path_cover_s,
+                        r.path_cover_l as path_cover_l,
+                        jsonb_array_elements_text(igdb_metadata -> 'franchises') as collection_name,
+                        'franchise' as collection_type
+                    FROM roms r
+                    WHERE igdb_metadata->'franchises' IS NOT NULL
+                ),
+                collection_collection AS (
+                    SELECT
+                        r.id as rom_id,
+                        r.path_cover_s as path_cover_s,
+                        r.path_cover_l as path_cover_l,
+                        jsonb_array_elements_text(igdb_metadata -> 'collections') as collection_name,
+                        'collection' as collection_type
+                    FROM roms r
+                    WHERE igdb_metadata->'collections' IS NOT NULL
+                ),
+                modes_collection AS (
+                    SELECT
+                        r.id as rom_id,
+                        r.path_cover_s as path_cover_s,
+                        r.path_cover_l as path_cover_l,
+                        jsonb_array_elements_text(igdb_metadata -> 'game_modes') as collection_name,
+                        'mode' as collection_type
+                    FROM roms r
+                    WHERE igdb_metadata->'game_modes' IS NOT NULL
+                ),
+                companies_collection AS (
+                    SELECT
+                        r.id as rom_id,
+                        r.path_cover_s as path_cover_s,
+                        r.path_cover_l as path_cover_l,
+                        jsonb_array_elements_text(igdb_metadata -> 'companies') as collection_name,
+                        'company' as collection_type
+                    FROM roms r
+                    WHERE igdb_metadata->'companies' IS NOT NULL
+                )
+                SELECT
+                    collection_name as name,
+                    collection_type as type,
+                    'Autogenerated ' || collection_type || ' collection' AS description,
+                    NOW() AS created_at,
+                    NOW() AS updated_at,
+                    array_to_json(array_agg(rom_id)) as rom_ids,
+                    array_to_json(array_agg(path_cover_s)) as path_covers_s,
+                    array_to_json(array_agg(path_cover_l)) as path_covers_l
```

Comment:
This differs from MySQL, where we use `DISTINCT` within the aggregation. I don't think that's needed as I don't see a way for CTEs to repeat `rom_id` values. We should have the same logic for both engines.

Reply:
i can't seem to figure out how to `DISTINCT` in postgres like i did in mysql 🤷🏼